frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardPage.tsx

```diff
@@ -0,0 +1,66 @@
+import type { WithRouterProps } from "react-router";
+
+import { useSyncURLSlug } from "metabase/dashboard/components/DashboardTabs/use-sync-url-slug";
+import {
+  useDashboardNav,
+  useDashboardUrlParams,
+  useRefreshDashboard,
+} from "metabase/dashboard/hooks";
+
+import { PublicOrEmbeddedDashboard } from "./PublicOrEmbeddedDashboard";
+import { usePublicDashboardEndpoints } from "./WithPublicDashboardEndpoints";
+
+export const PublicOrEmbeddedDashboardPage = (props: WithRouterProps) => {
+  const { location } = props;
+  const parameterQueryParams = props.location.query;
+
+  const { dashboardId } = usePublicDashboardEndpoints(props);
+
+  const { refreshDashboard } = useRefreshDashboard({
+    dashboardId,
+    parameterQueryParams,
+  });
+
+  const {
+    bordered,
+    hasNightModeToggle,
+    hideDownloadButton,
+    hideParameters,
+    isFullscreen,
+    isNightMode,
+    onNightModeChange,
+    refreshPeriod,
+    onFullscreenChange,
+    setRefreshElapsedHook,
+    onRefreshPeriodChange,
+    theme,
+    titled,
+    font,
+  } = useDashboardUrlParams({ location, onRefresh: refreshDashboard });
+
+  useDashboardNav({ isFullscreen });
+
+  useSyncURLSlug({ location });
+
+  return (
+    <PublicOrEmbeddedDashboard
+      dashboardId={dashboardId}
+      isFullscreen={isFullscreen}
+      refreshPeriod={refreshPeriod}
+      hideParameters={hideParameters}
```

Comment:
Interestingly in the initial migration I had errors here, that made me think that the HOC was eating some error types. The error is not present anymore after some other refactors done on master, but it's nice to see already that this is helping catch some subtle issues